### ISSUE TITLE
[bug][v9] Merchant View, rolltable to inventory

### ISSUE THREAD

**Issue by DanielBoettner** [jopeek/fvtt-loot-sheet-npc-5e#232]
**Describe the bug**
Users described an issue when the sheet is set to "merchant".
It seems that in the population of items within that view is bugged.

**To Reproduce**
1. Go to a NPC sheet
2. Click on GM Sidebar
3. Pick merchant type
4. Try to populate

**Expected behavior**
Items get added to the inventory with the results from the table.

**Comment by Tyreal74** [comment]
In addition to this which might be related, having the clear inventory checkbox enabled doesnt clear the inventory either.

**Comment by d1no1** [comment]
fvtt v8 latest
latest module version as of 01/01/2022
![brave_BzCO01NKrf](https://user-images.githubusercontent.com/88597883/147854005-1a935620-3e40-41bf-a680-6743154a62a4.png)
![brave_kOeaUFF1Av](https://user-images.githubusercontent.com/88597883/147854007-3ea3c679-0a11-474f-b748-23f972fcb36c.png)
![brave_hK4uZb207u](https://user-images.githubusercontent.com/88597883/147854008-3b9152dc-3251-4a55-842c-249eed779339.png)
the rolltable is in a compedium, and it has nested rolltables inside.
population of items fails even if the rolltable is in the rolltable directory instead of compendium.

i think the tables being nested is causing the issue, not sure if it relates to issue#232 but wanted to check before opening another one.

**Comment by d1no1** [comment]
solved by enabling "use better tables" please disregard!

**Comment by DanielBoettner** [comment]
Great to hear that it works with better rolltables.

I personally absolutely recommend the usage of BetterRolltables. For example it automatically creates scrolls from spell compendia. Next version will, for now convert any scene or actor to a 'portrait of actor' or 'map of scenename'.

I guess because of me always using BRT, I forgot to actually take a good look at the LS NPC internal loot population.

I will aim to bring a debug and test functionality without BRT for the next release. 

Throughout the code, there where quite some little bugs. And while I'm terribly sorry for an inconveniences after the refactoring. I still think it was the best course of Action to get the refactoring out there before v9 is in heavy use.

The module should v9 compatible with the next Version. And I will start to develop on v9 very soon.

I'll leave this issue open for a bit longer. For documentation.

But I sure hope that in the coming week most issues will be closed and reason to create new ones are mostly gone.

We already got some more features in mind. For when the bugs are gone.
First looks suggest that even selling to a merchant could be included soonish. 

I spoke to the Regions developer, the Quick Encounter dev and some more. For alot of them integration with the lootsheet offers  a very cool potential new feature set. I hope/think that devs will make greate use of the added API.

An update for the lootsheet generated chat messages will definitely ship with the next release. As well as coloring item rarity.

In other words, I'm positive 😅

**Comment by kirerellim** [comment]
I am having much the same issue, though I should note turning on better roll tables only partial fixes the problem for me. It only rolls up one item with quantity of one no matter what the settings are.

![image](https://user-images.githubusercontent.com/39993026/149559990-136f4be9-b6f1-4916-b836-8f61d4714e7e.png)

**Comment by LyricalStreetPoet** [comment]
I can confirm this is happening still.  Here are the issues I've seen.

When Checking Clear Inventory, you get this error (only if inventory exists):
![image](https://user-images.githubusercontent.com/32119569/150896722-32c7794a-eab4-487f-8a89-be3b40f550e2.png)

When selecting one of the default roll tables in a compendium there are no errors, but nothing gets populated:
![image](https://user-images.githubusercontent.com/32119569/150896981-e817d23e-73ed-4b1f-800a-65e708393a37.png)

When using BetterRollTables as stated by others, it works, but only one item of quantity 1 is created:
![image](https://user-images.githubusercontent.com/32119569/150897124-81b1fde6-ca31-44a8-a078-d3803af6a0dc.png)
